### PR TITLE
[Merged by Bors] - feat(ModelTheory/Order): Golf and implication instances on theories of orders

### PR DESCRIPTION
**Pull request by awainverse**
Shows a series of implications that models of stronger theories of orders are also models of weaker ones
Redefines some theories of orders using `insert`
Proves lemmas for handling theories defined with `insert`, simplifying a few proofs about order theories

---

- [x] depends on: #16338 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/ModelTheory/Complexity.lean
Comment (YaelDillies) [inline Mathlib/ModelTheory/Complexity.lean:418]: Isn't
```suggestion
  isUniversal_of_mem : ∀ ⦃φ⦄, φ ∈ T → φ.IsUniversal
```
the best of both worlds?

File: Mathlib/ModelTheory/Semantics.lean
Comment (YaelDillies) [inline Mathlib/ModelTheory/Semantics.lean:696]: This shouldn't be simp as `M` appears once the LHS and twice on the RHS 
```suggestion
theorem model_insert_iff {φ : L.Sentence} : M ⊨ insert φ T ↔ M ⊨ φ ∧ M ⊨ T := by
```

**Review by YaelDillies — COMMENTED** [review]
Can you please merge master to reduce the diff?

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Comment by awainverse** [comment]
I'm not sure why `merge-conflict` is still there after I've merged master and passed all checks

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
